### PR TITLE
DUOS-1238[risk=no]Add DAR view link to new DAC Chair Console 'unopened' DARs + text desc.

### PR DESCRIPTION
**Pull request by raejohanek**
SCOPE:
- Add links to review results page on non-open election statuses  (Unreviewed, Approved, Denied, Canceled) for the chair console
- Change -- to 'unreviewed' status for dars with no elections

ADDRESSES:
https://broadworkbench.atlassian.net/browse/DUOS-1238

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [ ] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/Routes.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ const Routes = (props) => (
     <AuthenticatedRoute path="/reviewed_cases" component={ReviewedCases} props={props}
       rolesAllowed={[USER_ROLES.admin, USER_ROLES.alumni]} />
     <AuthenticatedRoute path="/review_results/:referenceId" component={ReviewResults} props={props}
-      rolesAllowed={[USER_ROLES.admin, USER_ROLES.alumni]} />
+      rolesAllowed={[USER_ROLES.admin, USER_ROLES.alumni, USER_ROLES.chairperson]} />
```
Comment (rushtong) [inline src/Routes.js:109]: I think this should be open to members as well, but we might not have a code path for that yet.

**Comment by JVThomas** [comment]
`review_results` will always error for unreviewed DARs because the `Match.findMatch()` call needs an election id which they (unreviewed DARs) don't have, so you'll need to figure out how to make the Match call when needed.

Kapture 2021-05-13 at 13 28 47

File: src/pages/ReviewResults.js [hunk 2]
```diff
@@ -62,11 +62,13 @@ export default function ReviewResults(props) {
...
-      setMatchData(matchData);
-    } catch (e) {
-      Notifications.showError({ text: `Something went wrong trying to get match algorithm results. Error code: ${e.status}` });
+    if (!isNil(accessElection)) {
+      try {
+        matchData = await Match.findMatch(consent.consentId, accessElection.referenceId);
+        setMatchData(matchData);
+      } catch (e) {
+        Notifications.showError({text: `Something went wrong trying to get match algorithm results. Error code: ${e.status}`});
+      }
```
Comment (raejohanek) [inline src/pages/ReviewResults.js:75]: by adding the null check for accessElection this code works for unreviewed dars too

**Review by rushtong — CHANGES_REQUESTED** [review]
I think this PR goes a little further than the ticket requires. The ticket doesn't specifically request that "Unreviewed" link to the Review Results page, only that the string "- -" be replaced with "Unreviewed". I can see the logic in taking that step, but I think it might be scope creep. Additionally, if we do increase the scope to what this PR is trying to do, then ReviewResults will need some enhanced logic because when it's unreviewed, it's trying to get an election for a DAR that doesn't have one and there is a 404 in the network panel that we shouldn't be making.

**Comment by raejohanek** [comment]
> I think this PR goes a little further than the ticket requires. The ticket doesn't specifically request that "Unreviewed" link to the Review Results page, only that the string "- -" be replaced with "Unreviewed". I can see the logic in taking that step, but I think it might be scope creep. Additionally, if we do increase the scope to what this PR is trying to do, then ReviewResults will need some enhanced logic because when it's unreviewed, it's trying to get an election for a DAR that doesn't have one and there is a 404 in the network panel that we shouldn't be making.

I just talked to Jonathan about this ticket this morning and confirmed that this is what we want to do. I need to make the changes to the ticket to reflect that because it was a bit unclear. I'll take a look at the logic and try to simplify things for the unreviewed case

**Comment by raejohanek** [comment]
> > I think this PR goes a little further than the ticket requires. The ticket doesn't specifically request that "Unreviewed" link to the Review Results page, only that the string "- -" be replaced with "Unreviewed". I can see the logic in taking that step, but I think it might be scope creep. Additionally, if we do increase the scope to what this PR is trying to do, then ReviewResults will need some enhanced logic because when it's unreviewed, it's trying to get an election for a DAR that doesn't have one and there is a 404 in the network panel that we shouldn't be making.
> 
> I just talked to Jonathan about this ticket this morning and confirmed that this is what we want to do. I need to make the changes to the ticket to reflect that because it was a bit unclear. I'll take a look at the logic and try to simplify things for the unreviewed case

So I just added an optional param to pass to the review results page if the status is unreviewed, then if that optional param is present in the review results page it does not make the call to find the election, what do you think of this approach?

File: src/pages/ReviewResults.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ const SECTION = {
 
 export default function ReviewResults(props) {
   const darId = props.match.params.referenceId;
+  const status = props.match.params.referenceId;
```
Comment (rushtong) [inline src/pages/ReviewResults.js]: This looks like a copy paste error. Shouldn't it be looking for the `status` param?

File: src/pages/ReviewResults.js [hunk 4]
```diff
@@ -41,11 +42,14 @@ export default function ReviewResults(props) {
...
-      //so there is no vote information to display
+    //dars with unreviewed status will not have an election
+    if (isNil(status)) {
+      try {
+        accessElection = await Election.findElectionByDarId(darId);
+      } catch (error) {
+        //access election is null, this is expected for a dar with an unreviewed election status
+        //so there is no vote information to display
+      }
     }
```
Comment (rushtong) [inline src/pages/ReviewResults.js]: If we know that `Unreviewed` will have no election, why are we making the query at all? Could you wrap the call with `if (status !== 'Unreviewed')` ?

**Review by rushtong — CHANGES_REQUESTED** [review]
Looking better. See comments inline.

**Review by rushtong — APPROVED** [review]
Thanks for the changes. As fallout from this, I created https://broadworkbench.atlassian.net/browse/DUOS-1265.

**Pull request merged**